### PR TITLE
feat(codebuild): allow setting queued timeout

### LINKED ISSUE #11364: [codebuild] support LogsConfig and QueuedTimeoutInMinutes in L2 construct
The [CodeBuild L2 construct](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-codebuild.Project.html) does not support [LogsConfig](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-project.html#cfn-codebuild-project-logsconfig) and [QueuedTimeoutInMinutes](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-project.html#cfn-codebuild-project-queuedtimeoutinminutes).

### Use Case

Specify the `LogsConfig` and `QueuedTimeoutInMinutes` properties via L2 CodeBuild construct. Also those properties should be override when starting build via Step Functions integration. See https://github.com/aws/aws-cdk/pull/10356 for detail.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by ialford**
Closes #11364

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codebuild/README.md [hunk 1]
```diff
@@ -617,3 +617,25 @@ if (project.enableBatchBuilds()) {
   console.log('Batch builds were enabled');
 }
 ```
+
+## Timeouts
+
+There are two types of timeouts that can be set when creating your Project. The `timeout` property can be used to set an upper limit on how long your Project is able to run without being marked as completed. The default is 60 minutes. An example of overriding the default follows.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codebuild/README.md:623]: Can you split this paragraph into multiple lines on sentence boundaries? These files should be readable in their plain-text form as well as after being rendered by a Markdown engine.

```sugguestion
There are two types of timeouts that can be set when creating your Project.
The `timeout` property can be used to set an upper limit on how long your Project is able to run without being marked as failed.
The default is 60 minutes.
An example of overriding the default:
```

File: packages/@aws-cdk/aws-codebuild/README.md [hunk 2]
```diff
@@ -617,3 +617,25 @@ if (project.enableBatchBuilds()) {
...
+
+```ts
+import * as codebuild from '@aws-cdk/aws-codebuild';
+
+new codebuild.Project(stack, 'MyProject', {
+  timeout: Duration.minutes(90)
+};
+```
+
+The `queuedTimeout` property can be used to set an upper limit on how your Project remains queued to run. There is no default value for this property. As an example, to allow your Project to queue for up to thirty (30) minutes before the build fails, use the following code.
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codebuild/README.md:633]: Same comment here as above:

```suggestion
The `queuedTimeout` property can be used to set an upper limit on how your Project remains queued to run.
There is no default value for this property.
As an example, to allow your Project to queue for up to thirty (30) minutes before the build fails,
use the following code:
```

File: packages/@aws-cdk/aws-codebuild/README.md [hunk 3]
```diff
@@ -617,3 +617,25 @@ if (project.enableBatchBuilds()) {
...
+## Timeouts
+
+There are two types of timeouts that can be set when creating your Project. The `timeout` property can be used to set an upper limit on how long your Project is able to run without being marked as completed. The default is 60 minutes. An example of overriding the default follows.
+
+```ts
+import * as codebuild from '@aws-cdk/aws-codebuild';
+
+new codebuild.Project(stack, 'MyProject', {
+  timeout: Duration.minutes(90)
+};
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codebuild/README.md:630]: Missing the closing `)`:

```suggestion
});
```

File: packages/@aws-cdk/aws-codebuild/README.md [hunk 4]
```diff
@@ -617,3 +617,25 @@ if (project.enableBatchBuilds()) {
...
+```
+
+The `queuedTimeout` property can be used to set an upper limit on how your Project remains queued to run. There is no default value for this property. As an example, to allow your Project to queue for up to thirty (30) minutes before the build fails, use the following code.
+
+```ts
+import * as codebuild from '@aws-cdk/aws-codebuild';
+
+new codebuild.Project(stack, 'MyProject', {
+  queuedTimeout: Duration.minutes(30)
+};
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codebuild/README.md:640]: Missing the closing `)`:

```suggestion
});
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Looks great @ialford , thanks so much for the contribution!

A few documentation suggestions before we merge this puppy in!

**Comment by ialford** [comment]
@skinny85 - I have incorporated your suggestions. Let me know if there's anything else!

**Review by skinny85 — APPROVED** [review]
Thanks @ialford !

**Pull request merged**